AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -988,6 +988,119 @@ - (CGRect)convertRect:(CGRect)rect toNode:(ASDisplayNode *)node
   return CGRectApplyAffineTransform(rect, flattenedTransform);
 }
 
+#pragma mark - Layout Transition
+
+- (void)transitionLayoutWithAnimation:(BOOL)animated
+{
+  [self transitionLayoutToFit:_constrainedSize animated:animated];
```

Comment:
It doesn't need to be supported in the first version, but one of the key use cases to support his transitioning between different constrained sizes. Let's consider how we can integrate this into the API, Even if it does not work initially.